daemon/algod/api/algod.oas2.json

```diff
@@ -381,6 +398,10 @@
     "/v2/accounts/{address}/transactions/pending": {
       "get": {
         "description": "Get the list of pending transactions by address, sorted by priority, in decreasing order, truncated at the end at MAX. If MAX = 0, returns all pending transactions.\n",
+        "tags": [
+          "public",
+          "participating"
```

Comment:
similar to previous question, are these endpoints not available to nodes running non-part mode?

Reply:
Good question! In this PR I'm just separating APIs into groups based on potential "modes" of operation for a node, but in the [routers.go](https://github.com/algorand/go-algorand/pull/4714/files#diff-fe8c95a9c86fdebcfbe4fd568766817b3e97869806d569eef03357f1c8ffb6fb) file I'm registering all of them as normal.

In a followup PR the plan is to implement variations of [AlgorandFullNode](https://github.com/algorand/go-algorand/blob/master/node/node.go#L101) which have different subsets of functionality turned off/on. 

Currently the list I'm working with is:  
* Participating-- runs APIs tagged with `nonparticipating` and `participating`
* NonParticipating-- runs APIs tagged with `nonparticipating`
* Data-- runs APIs tagged with `nonparticipating` and `data`

This should allow us to more easily turn on/off the APIs based on the purpose of the node, but also it will allow us to make the node more lightweight in certain cases. For example, a non participating node would no longer run the agreement service, or have a txn pool, etc.

And in the case of the data mode, we can ensure that the user has full control over what round the node is on because the agreement service won't be running to try to advance the round. That's important when implementing sync round and delta APIs in #4658

Reply:
@Eric-Warehime Based on what I understand PR scope to be, I _think_ we can defer a decision until a subsequent PR.  My intent is to raise for light consideration because we're in the neighborhood.

I'm interested in selecting a different name for _data_ mode.  
* Off-the-cuff, I don't have a better suggestion (_NonParticipatingDeveloper_?).
* I think you suggested as much in a verbal:  Perhaps we can move away from `EnableDeveloperAPI` towards mode instead?  
  * I'm unfamiliar with the history, but perhaps it's sensible for the endpoints guarded by `EnableDeveloperAPI` to be non-participating _only_.
  * If we grouped additional routes under the _data_ umbrella, it might motivate a different name.
* Ultimately, if we don't have alternative names, I'm good as is.  If you feel _Data_ works well, I also won't push further.

Reply:
I'll echo @michaeldiamant question (and caveat that this is a good time to discuss, but out of scope for the PR). I'm not sure if "NonParticipatingDeveloper" makes sense, since dev-mode is a single-node instance it must be participating. 


Additionally, I had some of the same confusion as Shiqi. I think the names "participating" and "nonparticipating" would be user settings that describe a set of feature, and the tags here might be clearer if written in terms of the feature.

* `participating` - I think this is fine. There are also participation specific endpoints, so it maps well to the API.
* `non-participating` - What is the difference between this and `data` (or `ledger`)

Reply:
* `admin` - probably a better name than public/private. Maybe we don't even need `public`.

Reply:
w/r/t public vs. private, I'm open to changing to `admin` for private. The reason I liked `public` is that made the directory structure a bit more predictable.

For the other names, the difference between a `non-participating` node and a `data` or `ledger` (or whatever) node is for now solely the new set of APIs being added for retrieving deltas and managing the ledger's sync round. I think if we want to we can also include things like the `v2/teal/` endpoints. Also, given the work being done for new Debugger hooks, and expanding the capability of the node to provide fine grained txn tracing, it makes sense to me to put APIs exposing that data in this mode--we probably wouldn't want Txn tracers running on a normal `participating` or `nonparticipating` node.